client/src/main/java/io/mantisrx/server/master/client/MantisMasterClientApi.java

```diff
@@ -660,22 +677,37 @@ private Integer getSinkStageNumFromJsonResponse(String jobId, String response) t
                         .connect()
                         .flatMap((ObservableConnection<TextWebSocketFrame, TextWebSocketFrame> connection) -> connection.getInput()
                                 .map((TextWebSocketFrame webSocketFrame) -> webSocketFrame.text())))
-                .onErrorResumeNext(Observable.empty());
+                .onErrorResumeNext(Observable.empty())
+                .doOnCompleted(() -> {
+                  logger.info("Purging {} from job status cache", id);
+                  jobSchedulingInfoCache.remove(id);
```

Comment:
@codyrioux This change will affect mantisclient and hence all jobs as well. Should we move the caching up a layer to reduce potential impact?

Reply:
Up a layer as in directly inside Mantis API?

If so, already implemented. I thought we might want to do it here because multiple connections to jobs via the Mantis API (if they use different query params) will still cause multiple subscriptions to scheduling info.

Reply:
I see, yah in that case we could make this optional and default it to false, and then only enable it on mantis api first and then slowly roll out to jobs perhaps?

Reply:
yeah, given the blast radius, +1 for putting this behind a property so it can be disabled at runtime if we see any issues

Reply:
+1 on rollout. should we also add a unit test to test this behavior?